dask/dataframe/groupby.py

```diff
@@ -2711,7 +2732,26 @@ def rolling(self, window, min_periods=None, center=False, win_type=None, axis=0)
             axis=axis,
         )
 
-    def fillna(self, value=None, method=None, limit=None, axis=None):
+    def _normalize_axis(self, axis, method: str):
+        if PANDAS_GT_210 and axis is not no_default:
+            if axis in (0, "index"):
+                warnings.warn(
+                    f"The 'axis' keyword in {type(self).__name__}.{method} is deprecated and will "
+                    "be removed in a future version. Call without passing 'axis' instead.",
+                    FutureWarning,
+                )
+            else:
```

Comment:
I had assumed `(1, "columns", None)` based on [`DataFrame._validate_axis`](https://github.com/dask/dask/blob/ea913444dbcdbc43687ae581633f6b6bb0f87ea2/dask/dataframe/core.py#L5518), but I recognize that this list varies; I guess my underlying question is why we opt to flip pandas' approach here of specifying the deprecated `axis` argument in a finite set of cases, and the general deprecation warning otherwise:

https://github.com/pandas-dev/pandas/blob/3083ae932e0b122e963608fb25cf3df06d588b03/pandas/core/groupby/groupby.py#L980-L995

Reply:
`pandas` has some internal methods for normalizing to 0 or 1 for input `axis` values (i.e. `"columns"` --> 1). For example, not even when one specifies `axis="columns"` in `pandas`, the deprecation warnings that's emitted says `axis=1`

```python
In [6]: type(df)
Out[6]: pandas.core.frame.DataFrame

In [7]: df
Out[7]:
   A  B    C    D    E
0  1  3  NaN  4.0  6.0
1  1  4  3.0  NaN  NaN
2  2  3  NaN  5.0  7.0
3  2  4  NaN  NaN  NaN

In [8]: df.groupby("A").fillna(0, axis="columns")
<ipython-input-8-b27392e2513d>:1: FutureWarning: DataFrameGroupBy.fillna with axis=1 is deprecated and will be removed in a future version. Operate on the un-grouped DataFrame instead
  df.groupby("A").fillna(0, axis="columns")
Out[8]:
     B    C    D    E
0  3.0  0.0  4.0  6.0
1  4.0  3.0  0.0  0.0
2  3.0  0.0  5.0  7.0
3  4.0  0.0  0.0  0.0
```

In this PR we're not normalizing to integers and are instead using the user-provided `axis` value in the deprecation warning

```python
In [12]: ddf.groupby("A").fillna(0, axis="columns")
/Users/james/projects/dask/dask/dask/dataframe/groupby.py:2741: FutureWarning: DataFrameGroupBy.fillna with axis=columns is deprecated and will be removed in a future version. Operate on the un-grouped DataFrame instead
  warnings.warn(
Out[12]:
Dask DataFrame Structure:
                     B        C        D        E
npartitions=2
               float64  float64  float64  float64
                   ...      ...      ...      ...
                   ...      ...      ...      ...
Dask Name: droplevel, 8 graph layers
```

I think this is a bit more clear, but probably not a huge deal either way. Does that help clarify things? Or am I missing something?